dist/crawl_links.js

```diff
@@ -0,0 +1,65 @@
+/*
+Collects up to 20 same-origin in-viewport links from img, h[1-6] tags,
+sorted by size from largest to smallest.
+This is the seed for generating a list of URLs to crawl from the given page.
+*/
+const intersectRect = function (r1, r2) {
+    return !(r2.left > r1.right ||
+              r2.right < r1.left ||
+              r2.top > r1.bottom ||
+              r2.bottom < r1.top);
+};
+const sameOrigin = function(uri1, uri2){
+    try {
+        uri1 = new URL(uri1);
+        uri2 = new URL(uri2);
+        return uri1.origin == uri2.origin;
+    } catch(e) {
+    }
+    return false;
+}
+const viewport = {
+    left: 0,
+    top: 0,
+    right: (window.innerWidth || document.documentElement.clientWidth),
+    bottom: (window.innerHeight || document.documentElement.clientHeight)
+};
+const getLinks = function(visibleOnly){
+    let links = {};
+    const documentOrigin = window.location.href;
+    const elements = document.links;
+    for (let e of elements) {
+        try {
+            const rect = e.getBoundingClientRect();
+            const style = window.getComputedStyle(e);
+            const is_visible = e.offsetParent !== null &&
+                               style.visibility !== 'hidden' &&
+                               style.display !== 'none' &&
+                               intersectRect(rect, viewport);
```

Comment:
Annoyingly the editorconfig linter doesn't handle hanging indents like this when non-even number of indent spaces.

We could turn the indent checking off, but it is useful in other cases.

```suggestion
            const is_visible = e.offsetParent !== null &&
                                style.visibility !== 'hidden' &&
                                style.display !== 'none' &&
                                intersectRect(rect, viewport);
```